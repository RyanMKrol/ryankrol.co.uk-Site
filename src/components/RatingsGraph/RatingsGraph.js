import React, { Component } from 'react';
import { Radar } from 'react-chartjs-2';

import styles from './RatingsGraph.module.css';

/**
 * Component to display ratings data
 */
class RatingsGraph extends Component {
  /**
   * @class
   * @param {object} props data, and labels for the ratings component
   */
  constructor(props) {
    super();

    if (typeof props.data === 'undefined' || typeof props.labels === 'undefined') {
      throw new Error('Could not create RatingsGraph');
    }

    this.colour1 = Math.random() * 255;
    this.colour2 = Math.random() * 255;
    this.colour3 = Math.random() * 255;

    this.data = props.data;
    this.labels = props.labels;
  }

  /**
   * Render method
   *
   * @returns {React.Component} JSX MovieRatingsItem component
   */
  render() {
    return (
      <div className={styles.container}>
        <Radar
          data={{
            labels: this.labels,
            datasets: [
              {
                data: this.data,
                backgroundColor: `rgba(${this.colour1}, ${this.colour2}, ${this.colour3}, 0.2)`,
                borderColor: `rgb(${this.colour1}, ${this.colour2}, ${this.colour3})`,
              },
            ],
          }}
          options={{
            plugins: {
              legend: {
                display: false,
              },
            },
            scales: {
              r: {
                ticks: {
                  display: false,
                  count: 5,
                },
                backgroundColor: '#fff',
                angleLines: {
                  display: true,
                },
                suggestedMin: 0,
                suggestedMax: 100,
              },
            },
          }}
        />
      </div>
    );
  }
}

export default RatingsGraph;
