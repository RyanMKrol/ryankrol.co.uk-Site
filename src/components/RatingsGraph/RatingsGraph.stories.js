import React from 'react';

import DEFAULT_MOCK from './__mocks__/RatingsGraphMocks';

export default {
  title: 'Components/RatingsGraph',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the RatingsGraph
 */
export const normal = () => DEFAULT_MOCK;
