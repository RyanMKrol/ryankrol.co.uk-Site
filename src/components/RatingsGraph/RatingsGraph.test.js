import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import RatingsGraph from './RatingsGraph';

import DEFAULT_MOCK from './__mocks__/RatingsGraphMocks';

// mocking Math random to ensure colours are constant in snapshot
const mockMath = Object.create(global.Math);
mockMath.random = () => 0.5;
global.Math = mockMath;

it('renders RatingsGraph without issue', () => {
  const listensItem = shallow(DEFAULT_MOCK);
  expect(toJson(listensItem)).toMatchSnapshot();
});

it('RatingsGraph throws without data', () => {
  expect(() => mount(<RatingsGraph />)).toThrow();
});

it('RatingsGraph throws without labels', () => {
  expect(() => mount(<RatingsGraph data={{}} />)).toThrow();
});
