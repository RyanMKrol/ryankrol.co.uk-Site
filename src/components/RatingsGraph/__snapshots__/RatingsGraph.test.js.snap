// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RatingsGraph without issue 1`] = `
<ForwardRef
  data={
    Object {
      "datasets": Array [
        Object {
          "backgroundColor": "rgba(127.5, 127.5, 127.5, 0.2)",
          "borderColor": "rgb(127.5, 127.5, 127.5)",
          "data": Array [
            10,
            20,
            30,
            40,
            50,
          ],
        },
      ],
      "labels": Array [
        "Label1",
        "Label2",
        "Label3",
        "Label4",
        "Label5",
      ],
    }
  }
  options={
    Object {
      "maintainAspectRatio": false,
      "plugins": Object {
        "legend": Object {
          "display": false,
        },
      },
      "responsive": true,
      "scales": Object {
        "r": Object {
          "angleLines": Object {
            "display": true,
          },
          "backgroundColor": "#fff",
          "suggestedMax": 100,
          "suggestedMin": 0,
          "ticks": Object {
            "count": 5,
            "display": false,
          },
        },
      },
    }
  }
/>
`;
